core/src/main/scala/kafka/security/SecurityUtils.scala

```diff
@@ -0,0 +1,48 @@
+/**
+  * Licensed to the Apache Software Foundation (ASF) under one or more
+  * contributor license agreements.  See the NOTICE file distributed with
+  * this work for additional information regarding copyright ownership.
+  * The ASF licenses this file to You under the Apache License, Version 2.0
+  * (the "License"); you may not use this file except in compliance with
+  * the License.  You may obtain a copy of the License at
+  *
+  *    http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+
+package kafka.security
+
+import kafka.security.auth.{Acl, Operation, PermissionType, Resource, ResourceType}
+import org.apache.kafka.common.acl.{AccessControlEntry, AclBinding, AclBindingFilter}
+import org.apache.kafka.common.resource.{Resource => AdminResource, ResourceType => AdminResourceType}
+import org.apache.kafka.common.security.auth.KafkaPrincipal
+
+import scala.util.Try
+
+
+object SecurityUtils {
+
+  def convertToResourceAndAcl(filter: AclBindingFilter): Try[(Resource, Acl)] = {
+    for {
+      resourceType <- Try(ResourceType.fromJava(filter.resourceFilter.resourceType))
+      principal <- Try(KafkaPrincipal.fromString(filter.entryFilter.principal))
+      operation <- Try(Operation.fromJava(filter.entryFilter.operation))
+      permissionType <- Try(PermissionType.fromJava(filter.entryFilter.permissionType))
+      resource = Resource(resourceType, filter.resourceFilter.name)
+      acl = Acl(principal, permissionType, filter.entryFilter.host, operation)
+    } yield (resource, acl)
+  }
+
+  def convertToAclBinding(resource: Resource, acl: Acl): AclBinding = {
+    val adminResource = new AdminResource(AdminResourceType.fromString(resource.resourceType.toString), resource.name)
```

Comment:
This `fromString(toString)` stuff won't work because the Scala classes are camelcase, and the Java classes use underscores in the names.  We need to use the `toJava` methods here.

(KafkaPrincipal might be an exception in that it does work there)